### PR TITLE
Update order status colors

### PR DESCRIPTION
**Pull request by sarjon**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | See ticket.
| Type?         | improvement
| Category?     | IN
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | https://github.com/PrestaShop/PrestaShop/issues/15228
| How to test?  | After installing/upgrading PS, order colors should change. You must use the last version of autoupgrade 4.10.0

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/16104)

### CODE REVIEW

**Comment by sarjon** [comment]
Hey @TristanLDD, here's how new colors look like

colors

Legacy has this feature, if background is bright it makes text dark, otherwise text is white. Brightness is calculated using some formula.

File: install-dev/upgrade/sql/1.7.7.0.sql
Comment (sarjon) [inline install-dev/upgrade/sql/1.7.7.0.sql]: I saw methods are "registered" this way, I assume they are executed somehow? :thinking:

**Review by Progi1984 — DISMISSED** [review]
Need second review. I'm not sure for the call from SQL file.

File: install-dev/upgrade/php/ps_1770_update_order_status_colors.php
Comment (PierreRambaud) [inline install-dev/upgrade/php/ps_1770_update_order_status_colors.php:34]: Why not using const defined in config.inc.php?

Comment (sarjon) [reply install-dev/upgrade/php/ps_1770_update_order_status_colors.php:34]: Because of deprecation, see

https://github.com/PrestaShop/PrestaShop/blob/develop/config/config.inc.php#L242

Comment (PierreRambaud) [reply install-dev/upgrade/php/ps_1770_update_order_status_colors.php:34]: Oh, you're right, didn't saw it. Could we have class const at least? or is it too overkill?

File: install-dev/upgrade/sql/1.7.7.0.sql
Comment (PierreRambaud) [reply install-dev/upgrade/sql/1.7.7.0.sql]: yes, with this code you call the script `ps_1770_update_order_status_colors.php` and the function with the same name :)

Comment (sarjon) [reply install-dev/upgrade/sql/1.7.7.0.sql]: I'm confused, I don't know if that's cool or just a hack. :boom:

File: install-dev/upgrade/php/ps_1770_update_order_status_colors.php
Comment (sarjon) [reply install-dev/upgrade/php/ps_1770_update_order_status_colors.php:34]: Hmmm, I guess it would make sense to have class constants for payment status config names, but it's beyond the scope of this PR imo.

**Review by PierreRambaud — DISMISSED** [review]

**Comment by Robin-Fischer-PS** [comment]
Hi @jolelievre @matks !

Some statuses are not displayed correctly :

- Partial refund should be "green HEX #2ECC71"
- Waiting capture (if it's a real status) should be "dark blue HEX #34219E"
- Waiting for Local Payment Method should be "dark blue HEX #34219E"
- Waiting for PayPal Payment should be "dark blue HEX #34219E"
- Waiting for Credit Card Payment should be "dark blue HEX #34219E"

Also, with a 1-click Upgrade, it did not worked, don't really know why.

Thanks

**Comment by jolelievre** [comment]
Hi @Robin-Fischer-PS 

The problem here comes from the `ps_checkout` module, the order states you mentioned come from this module which insert them with his colors (matching legacy ones). The module should be updated to change these colors.

However I made some changes so now on upgrade almost all statuses are correctly updated, based on their parameters I was able to identify some of them and assigne the correct color. The only left one is `Partial refund` which can't be deduced and will have to be dealt with in the module.

I noticed an error on update on the menu, but it's not related to this PR, the problem comes the the new `enabled` field on Tabs, to test your back office correctly after the upgrade, either type the urls manually or you can update the database content manually and set the column `enabled` to 1 in all the `ps_tab` table.

I opened a dedicated issue for this problem: https://github.com/PrestaShop/PrestaShop/issues/16379

**Review by Progi1984 — DISMISSED** [review]

**Review by matks — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Comment by colinegin** [comment]
Hello @jolelievre and @Robin-Fischer-PS ,

FYI I've told @Darmona about these changes, he is going to apply the same changes in the PS Checkout module for more consistency. There is no ETA, but it will be done !

**Comment by jolelievre** [comment]
I believe this PR can't be tested before #16379 is merged, it will also need a rebase for upgrade scripts and besides the ending `;` is missing in the PHP script calls in the sql file

**Review by matks — APPROVED** [review]

**Comment by Robin-Fischer-PS** [comment]
Hi @jolelievre ,

With autoupgrade, the status colors are now OK (excepted Partial Refund as you said), so it's :heavy_check_mark: for QA !

Thanks

**Comment by matks** [comment]
Thank you @sarjon and @jolelievre

**Pull request merged**
